common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -786,19 +786,19 @@ static boolean isAndroid() {
     }
 
     private static boolean isAndroid0() {
-        boolean android;
-        try {
-            Class.forName("android.app.Application", false, getSystemClassLoader());
-            android = true;
-        } catch (Throwable ignored) {
-            // Failed to load the class uniquely available in Android.
-            android = false;
-        }
+        // Idea: Sometimes java binaries include Android classes on the classpath, even if it isn't actually Android.
+        // Rather than check if certain classes are present, just check the VM, which is tied to the JDK.
+
+        // Optional improvement: check if `android.os.Build.VERSION` is >= 24. On later versions of Android, the
+        // OpenJDK is used, which means `Unsafe` will actually work as expected.
 
-        if (android) {
+        // Android sets this property to Dalvik, regardless of whether it actually is.
+        String vmName = SystemPropertyUtil.get("java.vm.name");
+        boolean isAndroid = vmName.equals("Dalvik");
```

Comment:
@normanmaurer depending on security manager or jvm could SystemPropertyUtil.get("java.vm.name") return null? might be safer to use yoda style equals here.

Reply:
https://github.com/netty/netty/pull/7700

Reply:
hmm.. maybe not a bad idea. @johnou want to do a pr ?